### PR TITLE
Feature/add pre commit

### LINKED ISSUE #103: Add pre-commit hooks for automated code quality checks
Implement pre-commit hooks to automatically run formatting, linting, and basic checks before commits, reducing CI failures and improving developer experience.

### PR DESCRIPTION
**Pull request by t4kamura**
## Description

I have enabled pre-commit hooks for contributors.

[Lefthook](https://github.com/evilmartians/lefthook) can now be set up using make. `make install-lefthook`

The following checks will run during pre-commit:
* `make test`
* `make golangci-lint`
* Check for differences when running `go test ./internal/analyzers/govalid -update`
* Check for differences after running `go mod tidy` on each go.mod file

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [ ] Refactoring (no functional changes)

## Quality Checklist

Before submitting this pull request, ensure your contribution meets these quality standards:

### Code Quality
- [x] Code passes `make golangci-lint` without errors
- [x] All tests pass: `make test`
- [x] Code follows existing patterns and conventions
- [x] No hard-coded values (use constants or configuration)
- [x] Error messages are clear and actionable

### Testing Requirements
- [ ] Golden tests created and passing (`internal/analyzers/govalid/testdata/`)
- [ ] Unit tests implemented with boundary value testing (`test/unit/`)
- [ ] Benchmark tests comparing against popular validation libraries (`test/benchmark/`)
  - [ ] [go-playground/validator](https://github.com/go-playground/validator) (BenchmarkGoPlayground*)
  - [ ] [asaskevich/govalidator](https://github.com/asaskevich/govalidator) (BenchmarkGoValidator*)
  - [ ] [go-ozzo/ozzo-validation](https://github.com/go-ozzo/ozzo-validation) (BenchmarkOzzoValidation*)
  - [ ] [gookit/validate](https://github.com/gookit/validate) (BenchmarkGookitValidate*)
- [ ] All test files generated: `cd test && go generate`

### Performance Standards
- [ ] Zero allocations in validation logic (verified via benchmarks)
- [ ] Performance improvement over existing validation libraries (minimum 2x faster)
- [ ] Benchmark results added to `test/benchmark/README.md`

### Documentation
- [ ] README.md updated with new marker documentation
- [ ] Code includes appropriate comments and examples
- [ ] Validator usage examples provided

### Integration
- [ ] Validator scaffold generated with `make generate-validator MARKER=yourmarker`
- [ ] Registry files automatically updated
- [ ] Binary rebuilt and installed: `go install ./cmd/govalid/`

### Final Verification
- [ ] All GitHub Actions CI checks pass
- [x] No breaking changes to existing functionality
- [x] Contribution follows project license requirements

## Related Issues

Closes #103

### CODE REVIEW

File: lefthook.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      run: go mod tidy -diff
+      glob: "{go.mod,go.sum}"
+    gomod-tools:
+      root: tools
+      run: go mod tidy -diff
+      glob: "tools/{go.mod,go.sum}"
+    gomod-test:
+      root: test
+      run: go mod tidy -diff
+      glob: "test/{go.mod,go.sum}"
```
Comment (sivchari) [inline lefthook.yaml:25]:
```suggestion
      glob: "test/{go.mod,go.sum}"
      
```

File: README.md [hunk 2]
```diff
@@ -263,6 +263,16 @@ govalid consistently outperforms reflection-based validators by **5x to 44x**:
...
 
+### 🔧 Development Setup
+
+For contributors, install lefthook to enable pre-commit checks:
+
+```bash
+make install-lefthook
+```
+
+This will install and configure lefthook to run automatic checks before each commit, ensuring code quality and consistency.
```
Comment (sivchari) [inline README.md:274]:
```suggestion
Because of this, lefthook is installed, then the code-base would be checked automatically before each commit, ensuring code quality and consistency.
```

**Review by sivchari — APPROVED** [review]
Thx!
